sopel/cli/run.py

```diff
@@ -0,0 +1,658 @@
+    # All we should need here is a return value, but making this
+    # a return makes Sopel hang on ^C after it says "Closed!"
+    os.unlink(pid_file)
+    os._exit(0)
+
+
+def add_legacy_options(parser):
+    parser.add_argument("-d", '--fork', action="store_true",
+                        dest="daemonize", help="Daemonize Sopel")
+    parser.add_argument("-q", '--quit', action="store_true", dest="quit",
+                        help=(
+                            "Gracefully quit Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop` instead)"))
+    parser.add_argument("-k", '--kill', action="store_true", dest="kill",
+                        help=(
+                            "Kill Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop --kill` instead)"))
+    parser.add_argument("-r", '--restart', action="store_true", dest="restart",
+                        help=(
+                            "Restart Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel restart` instead)"))
+    parser.add_argument("-l", '--list', action="store_true",
+                        dest="list_configs",
+                        help="List all config files found")
+    parser.add_argument("-m", '--migrate', action="store_true",
+                        dest="migrate_configs",
+                        help="Migrate config files to the new format")
+    parser.add_argument('--quiet', action="store_true", dest="quiet",
+                        help="Suppress all output")
+    parser.add_argument('-w', '--configure-all', action='store_true',
+                        dest='wizard',
+                        help=(
+                            "Run the configuration wizard "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel configure` instead)"))
+    parser.add_argument('--configure-modules', action='store_true',
+                        dest='mod_wizard',
+                        help=(
+                            "Run the configuration wizard, but only for the "
+                            "module configuration options "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel configure --modules` instead)"))
+    parser.add_argument('-v', action="store_true",
+                        dest='version_legacy',
+                        help=(
+                            "Show version number and exit "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use -V/--version instead)"))
+    parser.add_argument('-V', '--version', action='store_true',
+                        dest='version',
+                        help='Show version number and exit')
```

Comment:
No deprecation note because this will be moved into `start` for Sopel 8? (Could be a good place for `# TODO`.)

Reply:
This option is one of the few that I don't know what to do for now. It's, like, `sopel start` don't need it, it makes little to no sense.

But `sopel --version` is useful. But maybe it should be replaced by `sopel version` instead?

So many questions, and I don't have an answer for now. Since the "legacy" mode is a big TODO by itself, I would argue that we should take care of it later, without adding more to it for now.

Reply:
Can the argument parser have a set of default flags with no associated action that falls through to a default subparser/action (`legacy` in 7, `start` in 8) if none of those flags is given?

Just spitballing, unless it's easy to move this argument out of the `legacy` parser right now. I agree that `sopel start` doesn't need this argument, and it should go somewhere else ideally.

Reply:
There is no easy way to do that, because if you define a set of arguments on the main parser object, you will be able to do that:

    $ sopel --global-options start --start-options
    $ sopel start --start-options
    $ sopel-start --global-options --start-options

But not that:

    $ sopel start --global-options --start-options

Which is quite confusing when you think about it. Add to the mix the fact that, in Py2.7, the sub-parser argument **is always required**, and you have a recipe for nightmare and headaches.

Reply:
That _would_ be confusing… :frowning:

I guess `sopel start` is going to have it after `legacy` goes away, then. Ah, well.